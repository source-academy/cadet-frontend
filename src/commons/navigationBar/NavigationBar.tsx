import { Alignment, Classes, Icon, Navbar, NavbarDivider, NavbarGroup } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import * as classNames from 'classnames';
import * as React from 'react';
import { NavLink } from 'react-router-dom';

import { Role } from 'src/commons/application/ApplicationTypes';
import Dropdown from '../dropdown/DropdownComponent';

type NavigationBarProps = DispatchProps & StateProps;

type DispatchProps = {
  handleLogOut: () => void;
};

type StateProps = {
  role?: Role;
  title: string;
  name?: string;
};

const NavigationBar: React.SFC<NavigationBarProps> = props => (
  <Navbar className={classNames('NavigationBar', 'primary-navbar', Classes.DARK)}>
    <NavbarGroup align={Alignment.LEFT}>
      <NavLink
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        to="/playground"
      >
        <Icon icon={IconNames.CODE} />
        <div className="navbar-button-text hidden-xs hidden-sm">Source Academy Playground</div>
      </NavLink>
    </NavbarGroup>

    <NavbarGroup align={Alignment.RIGHT}>
      <NavLink
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        to="/contributors"
      >
        <Icon icon={IconNames.HEART} />
        <div className="navbar-button-text hidden-xs hidden-sm">Contributors</div>
      </NavLink>

      <div className="visible-xs">
        <NavbarDivider className="thin-divider" />
      </div>
      <div className="hidden-xs">
        <NavbarDivider className="default-divider" />
      </div>

      <Dropdown handleLogOut={props.handleLogOut} name={props.name} />
    </NavbarGroup>
  </Navbar>
);

export default NavigationBar;
