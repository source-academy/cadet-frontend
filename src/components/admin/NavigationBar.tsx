import * as React from 'react'

import { Alignment, Icon, Navbar, NavbarGroup } from '@blueprintjs/core'
import { NavLink } from 'react-router-dom'

const NavigationBar: React.SFC<{}> = () => (
  <Navbar className="NavigationBar">
    <NavbarGroup align={Alignment.LEFT}>
      <NavLink
        to="/admin/announcements"
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
      >
        <Icon icon="feed" />
        Announcements
      </NavLink>
      <NavLink
        to="/admin/assessments"
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
      >
        <Icon icon="tick" />
        Assessments
      </NavLink>
      <NavLink
        to="/admin/materials"
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
      >
        <Icon icon="folder-open" />
        Materials
      </NavLink>
    </NavbarGroup>
  </Navbar>
)

export default NavigationBar
