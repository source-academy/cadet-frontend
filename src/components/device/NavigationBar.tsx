import { Alignment, Icon, Navbar, NavbarGroup } from '@blueprintjs/core'
import { IconNames } from '@blueprintjs/icons'
import * as React from 'react'
import { NavLink } from 'react-router-dom'

const NavigationBar: React.SFC<{}> = () => (
  <Navbar className="NavigationBar">
    <NavbarGroup align={Alignment.LEFT}>
      <NavLink
        to="/device/missions"
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
      >
        <Icon icon={IconNames.FLAME} />
        Missions
      </NavLink>
      <NavLink
        to="/device/sidequests"
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
      >
        <Icon icon={IconNames.LIGHTBULB} />
        Sidequests
      </NavLink>
      <NavLink
        to="/device/paths"
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
      >
        <Icon icon={IconNames.PREDICTIVE_ANALYSIS} />
        Paths
      </NavLink>
      <NavLink
        to="/device/contests"
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
      >
        <Icon icon={IconNames.COMPARISON} />
        Contests
      </NavLink>
    </NavbarGroup>

    <NavbarGroup align={Alignment.RIGHT}>
      <NavLink
        to="/device/announcements"
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
      >
        <Icon icon={IconNames.FEED} />
        Announcements
      </NavLink>
      <NavLink
        to="/device/materials"
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
      >
        <Icon icon={IconNames.FOLDER_OPEN} />
        Materials
      </NavLink>
      <NavLink
        to="/device/status"
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
      >
        <Icon icon={IconNames.USER} />
        Profile
      </NavLink>
    </NavbarGroup>
  </Navbar>
)

export default NavigationBar
