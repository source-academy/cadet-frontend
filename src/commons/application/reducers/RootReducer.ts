import { connectRouter } from 'connected-react-router';
import { History } from 'history';
import { combineReducers } from 'redux';

import { AcademyReducer as academy } from '../../../features/academy/AcademyReducer';
import { DashboardReducer as dashboard } from '../../../features/dashboard/DashboardReducer';
import { PlaygroundReducer as playground } from '../../../features/playground/PlaygroundReducer';
import { WorkspaceReducer as workspaces } from '../../workspace/WorkspaceReducer';
import { ApplicationReducer as application } from '../ApplicationReducer';
import { SessionsReducer as session } from './SessionsReducer';

const createRootReducer = (history: History) =>
  combineReducers({
    router: connectRouter(history),
    academy,
    application,
    dashboard,
    playground,
    session,
    workspaces
  });

export default createRootReducer;
