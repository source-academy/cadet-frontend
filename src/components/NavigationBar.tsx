import {
  Alignment,
  Icon,
  Navbar,
  NavbarDivider,
  NavbarGroup,
  NavbarHeading
} from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import * as React from 'react';
import { NavLink } from 'react-router-dom';

import { Role } from '../reducers/states';
import { LINKS } from '../utils/constants';
import Dropdown from './dropdown';

export interface INavigationBarProps {
  handleLogOut: () => void;
  role?: Role;
  title: string;
  name?: string;
}

const NavigationBar: React.SFC<INavigationBarProps> = props => (
  <Navbar className="NavigationBar primary-navbar pt-dark">
    <NavbarGroup align={Alignment.LEFT}>
      <NavLink
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
        to="/academy"
      >
        <Icon icon={IconNames.SYMBOL_DIAMOND} />
        <NavbarHeading className="hidden-xs">Source Academy</NavbarHeading>
      </NavLink>

      <NavLink
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
        target="_blank"
        to={LINKS.IVLE}
      >
        <Icon icon={IconNames.BOOK} />
        <div className="navbar-button-text hidden-xs">News &amp; Material</div>
      </NavLink>

      <NavLink
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
        target="_blank"
        to={LINKS.PIAZZA}
      >
        <Icon icon={IconNames.CHAT} />
        <div className="navbar-button-text hidden-xs">Forum</div>
      </NavLink>
    </NavbarGroup>

    <NavbarGroup align={Alignment.RIGHT}>
      <NavLink
        activeClassName="pt-active"
        className="NavigationBar__link pt-button pt-minimal"
        to="/playground"
      >
        <Icon icon={IconNames.CODE} />
        <div className="navbar-button-text hidden-xs">Playground</div>
      </NavLink>

      <div className="visible-xs">
        <NavbarDivider className="thin-divider" />
      </div>
      <div className="hidden-xs">
        <NavbarDivider className="default-divider" />
      </div>

      <Dropdown handleLogOut={props.handleLogOut} name={props.name} />
    </NavbarGroup>
  </Navbar>
);

export default NavigationBar;
