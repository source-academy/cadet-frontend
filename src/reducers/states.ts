import { Context } from 'js-slang'
import { SourceError } from 'js-slang/dist/types'

import { WorkspaceLocation, WorkspaceLocations } from '../actions/workspaces'
import { Grading, GradingOverview } from '../components/academy/grading/gradingShape'
import { Announcement } from '../components/Announcements'
import {
  ExternalLibraryName,
  ExternalLibraryNames,
  IAssessment,
  IAssessmentOverview
} from '../components/assessment/assessmentShape'
import { HistoryHelper } from '../utils/history'
import { createContext } from '../utils/slangHelper'

export interface IState {
  readonly academy: IAcademyState
  readonly application: IApplicationState
  readonly playground: IPlaygroundState
  readonly session: ISessionState
  readonly workspaces: IWorkspaceManagerState
}

export interface IAcademyState {
  readonly gameCanvas?: HTMLCanvasElement
}

export interface IApplicationState {
  readonly title: string
  readonly environment: ApplicationEnvironment
}

export interface IPlaygroundState {
  readonly queryString?: string
}

interface IAssessmentWorkspace extends IWorkspaceState {
  readonly currentAssessment?: number
  readonly currentQuestion?: number
  readonly hasUnsavedChanges: boolean
}

interface IGradingWorkspace extends IWorkspaceState {
  readonly currentSubmission?: number
  readonly currentQuestion?: number
  readonly hasUnsavedChanges: boolean
}

export interface IPlaygroundWorkspace extends IWorkspaceState {
  readonly playgroundExternal: ExternalLibraryName
}

export interface IWorkspaceManagerState {
  readonly assessment: IAssessmentWorkspace
  readonly grading: IGradingWorkspace
  readonly playground: IPlaygroundWorkspace
}

export interface IWorkspaceState {
  readonly context: Context
  readonly editorValue: string | null
  readonly editorWidth: string
  readonly isEditorAutorun: boolean
  readonly isRunning: boolean
  readonly output: InterpreterOutput[]
  readonly replHistory: ReplHistory
  readonly replValue: string
  readonly sideContentActiveTab: number
  readonly sideContentHeight?: number
  readonly globals: Array<[string, any]>
}

export interface ISessionState {
  readonly accessToken?: string
  readonly assessmentOverviews?: IAssessmentOverview[]
  readonly assessments: Map<number, IAssessment>
  readonly announcements?: Announcement[]
  readonly grade: number
  readonly gradingOverviews?: GradingOverview[]
  readonly gradings: Map<number, Grading>
  readonly historyHelper: HistoryHelper
  readonly maxGrade: number
  readonly maxXp: number
  readonly refreshToken?: string
  readonly role?: Role
  readonly story?: Story
  readonly name?: string
  readonly xp: number
}

type ReplHistory = {
  browseIndex: null | number // [0, 49] if browsing, else null
  records: string[]
}

export const maxBrowseIndex = 50

export type Story = {
  story: string
  playStory: boolean
}

/**
 * An output while the program is still being run in the interpreter. As a
 * result, there are no return values or SourceErrors yet. However, there could
 * have been calls to display (console.log) that need to be printed out.
 */
export type RunningOutput = {
  type: 'running'
  consoleLogs: string[]
}

/**
 * An output which reflects the program which the user had entered. Not a true
 * Output from the interpreter, but simply there to let he user know what had
 * been entered.
 */
export type CodeOutput = {
  type: 'code'
  value: string
}

/**
 * An output which represents a program being run successfully, i.e. with a
 * return value at the end. A program can have either a return value, or errors,
 * but not both.
 */
export type ResultOutput = {
  type: 'result'
  value: any
  consoleLogs: string[]
  runtime?: number
  isProgram?: boolean
}

/**
 * An output which represents a program being run unsuccessfully, i.e. with
 * errors at the end. A program can have either a return value, or errors, but
 * not both.
 */
export type ErrorOutput = {
  type: 'errors'
  errors: SourceError[]
  consoleLogs: string[]
}

export type InterpreterOutput = RunningOutput | CodeOutput | ResultOutput | ErrorOutput

export enum ApplicationEnvironment {
  Development = 'development',
  Production = 'production',
  Test = 'test'
}

export enum Role {
  Student = 'student',
  Staff = 'staff',
  Admin = 'admin'
}

/**
 * Defines what chapters are available for usage.
 * For external libraries, see externalLibraries.ts
 */
export const sourceChapters = [1, 2, 3, 4]
const latestSourceChapter = sourceChapters.slice(-1)[0]

const currentEnvironment = (): ApplicationEnvironment => {
  switch (process.env.NODE_ENV) {
    case 'development':
      return ApplicationEnvironment.Development
    case 'production':
      return ApplicationEnvironment.Production
    default:
      return ApplicationEnvironment.Test
  }
}

export const defaultAcademy: IAcademyState = {
  gameCanvas: undefined
}

export const defaultApplication: IApplicationState = {
  title: 'Cadet',
  environment: currentEnvironment()
}

export const defaultPlayground: IPlaygroundState = {}

export const defaultEditorValue = '// Type your program in here!'

/**
 * Create a default IWorkspaceState for 'resetting' a workspace.
 * Takes in parameters to set the js-slang library and chapter.
 *
 * @param location the location of the workspace, used for context
 */
export const createDefaultWorkspace = (location: WorkspaceLocation): IWorkspaceState => ({
  context: createContext<WorkspaceLocation>(latestSourceChapter, [], location),
  editorValue: location === WorkspaceLocations.playground ? defaultEditorValue : null,
  editorWidth: '50%',
  output: [],
  replHistory: {
    browseIndex: null,
    records: []
  },
  replValue: '',
  sideContentActiveTab: 0,
  globals: [],
  isEditorAutorun: false,
  isRunning: false
})

export const defaultComments = 'Comments **here**. Use `markdown` if you ~~are cool~~ want!'

export const defaultWorkspaceManager: IWorkspaceManagerState = {
  assessment: {
    ...createDefaultWorkspace(WorkspaceLocations.assessment),
    currentAssessment: undefined,
    currentQuestion: undefined,
    hasUnsavedChanges: false
  },
  grading: {
    ...createDefaultWorkspace(WorkspaceLocations.grading),
    currentSubmission: undefined,
    currentQuestion: undefined,
    hasUnsavedChanges: false
  },
  playground: {
    ...createDefaultWorkspace(WorkspaceLocations.playground),
    playgroundExternal: ExternalLibraryNames.NONE
  }
}

export const defaultSession: ISessionState = {
  accessToken: undefined,
  announcements: [
    {
      author: 'Aministrator',
      title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
      content:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam non tellus non ligula consectetur feugiat a at mauris. Ut sagittis, urna id suscipit dictum, ipsum diam sollicitudin enim, sed ultricies diam turpis vel libero. Maecenas finibus nisl ac lobortis ultricies. Integer molestie urna vitae nisi pharetra porttitor. Vestibulum massa diam, tristique quis ante a, posuere placerat magna. Pellentesque at lobortis purus, vitae imperdiet diam. Nulla eu rutrum neque. Aliquam efficitur consectetur ullamcorper. Pellentesque ultricies, diam ut vestibulum pellentesque, metus arcu laoreet ex, at mattis quam est non neque. Nam velit ipsum, posuere non porttitor commodo, lobortis in urna. Nulla facilisi. Donec mollis id nibh a luctus. Mauris vitae orci id velit pulvinar pellentesque non ut sapien. Curabitur eu consequat lorem. Proin pretium blandit scelerisque. Morbi ultricies, tellus non posuere pretium, felis sem convallis magna, ut sagittis elit felis et sem. Aliquam auctor suscipit condimentum. Nam posuere nulla dolor, in maximus risus feugiat vel. Phasellus vestibulum odio nec leo vehicula condimentum. Ut et semper libero. Cras fermentum mauris quis ex sodales, sit amet hendrerit augue lobortis. Maecenas eu dapibus enim, nec auctor est. Quisque quis erat erat. Curabitur sed rutrum felis, non venenatis diam. Fusce maximus rhoncus neque, in maximus velit semper eu. Fusce tempus lorem ut sodales pulvinar.',
      pinned: true
    }
  ],
  assessments: new Map<number, IAssessment>(),
  assessmentOverviews: undefined,
  grade: 0,
  gradingOverviews: undefined,
  gradings: new Map<number, Grading>(),
  historyHelper: {
    lastAcademyLocations: [null, null],
    lastGeneralLocations: [null, null]
  },
  maxGrade: 0,
  maxXp: 0,
  refreshToken: undefined,
  name: undefined,
  xp: 0
}

export const defaultState: IState = {
  academy: defaultAcademy,
  application: defaultApplication,
  playground: defaultPlayground,
  session: defaultSession,
  workspaces: defaultWorkspaceManager
}
