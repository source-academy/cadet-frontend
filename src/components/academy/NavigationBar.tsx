import { Alignment, Classes, Icon, Navbar, NavbarGroup } from '@blueprintjs/core';
import { IconNames } from '@blueprintjs/icons';
import * as classNames from 'classnames';
import * as React from 'react';
import { NavLink } from 'react-router-dom';

import { Role } from '../../reducers/states';
import { assessmentCategoryLink } from '../../utils/paramParseHelpers';
import { AssessmentCategories } from '../assessment/assessmentShape';

import NotificationBadge from '../../containers/notification/NotificationBadge';
import { filterNotificationsByType } from '../notification/NotificationHelpers';

import DefaultChapter from '../../containers/DefaultChapterContainer';

type OwnProps = {
  role: Role;
};

const NavigationBar: React.SFC<OwnProps> = props => (
  <Navbar className="NavigationBar secondary-navbar">
    <NavbarGroup align={Alignment.LEFT}>
      <NavLink
        to={`/academy/${assessmentCategoryLink(AssessmentCategories.Mission)}`}
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
      >
        <Icon icon={IconNames.FLAME} />
        <div className="navbar-button-text hidden-xs">Missions</div>
        <NotificationBadge
          notificationFilter={filterNotificationsByType(AssessmentCategories.Mission)}
          disableHover={true}
        />
      </NavLink>

      <NavLink
        to={`/academy/${assessmentCategoryLink(AssessmentCategories.Sidequest)}`}
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
      >
        <Icon icon={IconNames.LIGHTBULB} />
        <div className="navbar-button-text hidden-xs">Quests</div>
        <NotificationBadge
          notificationFilter={filterNotificationsByType(AssessmentCategories.Sidequest)}
          disableHover={true}
        />
      </NavLink>

      <NavLink
        to={`/academy/${assessmentCategoryLink(AssessmentCategories.Path)}`}
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
      >
        <Icon icon={IconNames.PREDICTIVE_ANALYSIS} />
        <div className="navbar-button-text hidden-xs">Paths</div>
        <NotificationBadge
          notificationFilter={filterNotificationsByType(AssessmentCategories.Path)}
          disableHover={true}
        />
      </NavLink>

      <NavLink
        to={`/academy/${assessmentCategoryLink(AssessmentCategories.Contest)}`}
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
      >
        <Icon icon={IconNames.COMPARISON} />
        <div className="navbar-button-text hidden-xs">Contests</div>
        <NotificationBadge
          notificationFilter={filterNotificationsByType(AssessmentCategories.Contest)}
          disableHover={true}
        />
      </NavLink>

      <NavLink
        to={`/academy/${assessmentCategoryLink(AssessmentCategories.Practical)}`}
        activeClassName={Classes.ACTIVE}
        className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
      >
        <Icon icon={IconNames.MANUAL} />
        <div className="navbar-button-text hidden-xs">Practical</div>
      </NavLink>
    </NavbarGroup>
    {props.role === Role.Admin || props.role === Role.Staff ? (
      <NavbarGroup align={Alignment.RIGHT}>
        <DefaultChapter />

        <NavLink
          to={'/academy/dashboard'}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon="globe" />
          <div className="navbar-button-text hidden-xs">Dashboard</div>
        </NavLink>

        <NavLink
          to={'/academy/sourcereel'}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.MOBILE_VIDEO} />
          <div className="navbar-button-text hidden-xs">Sourcereel</div>
        </NavLink>

        <NavLink
          to={'/academy/material'}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.CLOUD_UPLOAD} />
          <div className="navbar-button-text hidden-xs">Upload</div>
        </NavLink>

        <NavLink
          to={'/academy/grading'}
          activeClassName={Classes.ACTIVE}
          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
        >
          <Icon icon={IconNames.ENDORSED} />
          <div className="navbar-button-text hidden-xs">Grading</div>
          <NotificationBadge
            notificationFilter={filterNotificationsByType('Grading')}
            disableHover={true}
          />
        </NavLink>
      </NavbarGroup>
    ) : null}
  </Navbar>
);

export default NavigationBar;
