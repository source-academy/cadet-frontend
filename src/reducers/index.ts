import { reducer as academy } from './academy';
import { reducer as application } from './application';
import { reducer as playground } from './playground';
import { reducer as session } from './session';
import { reducer as workspaces } from './workspaces';

export default {
  academy,
  application,
  playground,
  session,
  workspaces
};
