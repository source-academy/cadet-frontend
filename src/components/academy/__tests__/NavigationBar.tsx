import { shallow } from 'enzyme';
import * as React from 'react';

import { Role } from '../../../reducers/states';
import NavigationBar from '../NavigationBar';

test('Grading NavLink does NOT renders for Role.Student', () => {
  const props = { role: Role.Student };
  const tree = shallow(<NavigationBar {...props} />);
  expect(tree.debug()).toMatchSnapshot();
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: 'academy/grading' }).exists())
  ).toHaveLength(0);
});

test('Grading NavLink renders for Role.Staff', () => {
  const props = { role: Role.Staff };
  const tree = shallow(<NavigationBar {...props} />);
  expect(tree.debug()).toMatchSnapshot();
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/grading' }).exists())
  ).toHaveLength(1);
});

test('Grading NavLink renders for Role.Admin', () => {
  const props = { role: Role.Admin };
  const tree = shallow(<NavigationBar {...props} />);
  expect(tree.debug()).toMatchSnapshot();
  expect(
    tree.filterWhere(shallowTree => shallowTree.find({ to: '/academy/grading' }).exists())
  ).toHaveLength(1);
});
