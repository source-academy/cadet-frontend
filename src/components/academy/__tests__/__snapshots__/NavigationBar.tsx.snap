// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grading NavLink does NOT renders for Role.Student 1`] = `
"<Blueprint2.Navbar className=\\"NavigationBar secondary-navbar\\">
  <Blueprint2.NavbarGroup align=\\"left\\">
    <NavLink to=\\"/academy/missions\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"flame\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Missions
      </div>
    </NavLink>
    <NavLink to=\\"/academy/quests\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"lightbulb\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Quests
      </div>
    </NavLink>
    <NavLink to=\\"/academy/paths\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"predictive-analysis\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Paths
      </div>
    </NavLink>
    <NavLink to=\\"/academy/contests\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"comparison\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Contests
      </div>
    </NavLink>
  </Blueprint2.NavbarGroup>
</Blueprint2.Navbar>"
`;

exports[`Grading NavLink renders for Role.Admin 1`] = `
"<Blueprint2.Navbar className=\\"NavigationBar secondary-navbar\\">
  <Blueprint2.NavbarGroup align=\\"left\\">
    <NavLink to=\\"/academy/missions\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"flame\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Missions
      </div>
    </NavLink>
    <NavLink to=\\"/academy/quests\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"lightbulb\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Quests
      </div>
    </NavLink>
    <NavLink to=\\"/academy/paths\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"predictive-analysis\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Paths
      </div>
    </NavLink>
    <NavLink to=\\"/academy/contests\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"comparison\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Contests
      </div>
    </NavLink>
  </Blueprint2.NavbarGroup>
  <Blueprint2.NavbarGroup align=\\"right\\">
    <NavLink to=\\"/academy/grading\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"endorsed\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Grading
      </div>
    </NavLink>
  </Blueprint2.NavbarGroup>
</Blueprint2.Navbar>"
`;

exports[`Grading NavLink renders for Role.Staff 1`] = `
"<Blueprint2.Navbar className=\\"NavigationBar secondary-navbar\\">
  <Blueprint2.NavbarGroup align=\\"left\\">
    <NavLink to=\\"/academy/missions\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"flame\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Missions
      </div>
    </NavLink>
    <NavLink to=\\"/academy/quests\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"lightbulb\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Quests
      </div>
    </NavLink>
    <NavLink to=\\"/academy/paths\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"predictive-analysis\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Paths
      </div>
    </NavLink>
    <NavLink to=\\"/academy/contests\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"comparison\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Contests
      </div>
    </NavLink>
  </Blueprint2.NavbarGroup>
  <Blueprint2.NavbarGroup align=\\"right\\">
    <NavLink to=\\"/academy/grading\\" activeClassName=\\"pt-active\\" className=\\"NavigationBar__link pt-button pt-minimal\\" aria-current=\\"page\\">
      <Blueprint2.Icon icon=\\"endorsed\\" />
      <div className=\\"navbar-button-text hidden-xs\\">
        Grading
      </div>
    </NavLink>
  </Blueprint2.NavbarGroup>
</Blueprint2.Navbar>"
`;
